delivery_carrier_label_postlogistics/wizards/postlogistics_oauth.py

```diff
@@ -0,0 +1,101 @@
+# Copyright 2020 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+
+import json
+import requests
+
+from odoo import models, fields, api
+
+
+class PostlogisticsAuth(models.TransientModel):
+    _name = 'postlogistics.auth'
+    _description = 'Postlogistics Auth'
+
+    client_id = fields.Char(
+        string='Client ID',
+    )
+
+    client_secret = fields.Char(
+        string='Client Secret',
+    )
+
+    state = fields.Selection([
+        ('todo', 'OAuth Config'),
+        ('done', 'Complete')
+    ], default='todo')
+
+    @api.model
+    def default_get(self, field_list):
+        res = super().default_get(field_list)
+
+        # Get the client credentials token
+        icp = self.env['ir.config_parameter']
+        client_id = icp.get_param('postlogistics.oauth.client_id')
+        client_secret = icp.get_param('postlogistics.oauth.client_secret')
+        if client_id and client_secret:
+            res['client_id'] = client_id
+            res['client_secret'] = client_secret
+            res['state'] = 'done'
+
+        return res
+
+    @api.multi
+    def generate_access_token(self):
+        self.ensure_one()
+
+        authentication_url = 'https://wedecint.post.ch/WEDECOAuth/token'
```

Comment:
may be define this part in the global variable or in settings

Reply:
I will move it in settings in case of change, but I'm pretty sure it will never changed ;)